src/js/providers/flash.js

```diff
@@ -341,19 +323,8 @@ define([
                         this.trigger(e.type);
                     }, this);
 
-                    _swf.on(events.JWPLAYER_MEDIA_BEFORECOMPLETE, function(e){
-                        _beforecompleted = true;
+                    _swf.on(events.JWPLAYER_MEDIA_BEFORECOMPLETE, function(e) {
```

Comment:
What if providers just fire 'complete' and the model turns it into 'beforeComplete'? Then you don't have to handle 'beforeComplete' in instream.